ledger/src/blockstore.rs

```diff
@@ -3808,21 +3846,44 @@ where
     Ok(())
 }
 
+// Update the meta for a slot whose parent is known to be connected.
+// Returns a bool indicating whether this slot is now connected which would
+// indicate that it's children are also in need of update.
+fn update_child_from_connected_parent(meta: &mut SlotMeta) -> bool {
+    // If this slot is already connected, its' children should
+    // also already be connected so cut the traversal here.
+    if meta.is_connected() {
+        return false;
+    }
+    meta.set_parent_connected();
+
+    // This slot's parent is known to be connected, so if
+    // it is also full, then the slot is now connected.
+    let is_full = meta.is_full();
+    if is_full {
+        meta.set_connected();
```

Comment:
I feel this call should only set the bit (`01`) that only represents part of the full connectivity, such as `set_completed()` or something.  And this bit together with the parent_connected() (`10`) means connected (`11`).

Reply:
> I feel this call should only set the bit (01) that only represents part of the full connectivity, such as set_completed() or something. And this bit together with the parent_connected() (10) means connected (11).

We can only set connected iff parent_connected has previously been set. So, I think the scenario you described is an invalid one. Ie, we'll only set connected `(x1)` if parent_connected `(1y)` has already been set. Thus, it must be the case that `x = 1`. Additionally `y = 0` will always be true because we can't be connected before our parent was